include/zim/suggestion_iterator.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2021 Maneesh P M <manu.pm55@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * is provided AS IS, WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, and
+ * NON-INFRINGEMENT.  See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ */
+
+#ifndef ZIM_SUGGESTION_ITERATOR_H
+#define ZIM_SUGGESTION_ITERATOR_H
+
+#include "archive.h"
+
+namespace zim
+{
+class SuggestionResultSet;
+class SuggestionItem;
+class SearchIterator;
+
+class SuggestionIterator
```

Comment:
We probably want to SuggestionIterator inheriting `std::iterator` (as for SearchIterator) :
`class SuggestionIterator : public std::iterator<std::bidirection_iterator_tag, SuggestionItem>`

Reply:
Just to be sure, we just need to change the class definition to what you have mentioned right?

Reply:
Yes. It should be enough (or easy to fix raised errors).